pkg/noderesourcetopology/objects.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+func makePodByResourceList(resources *corev1.ResourceList) *corev1.Pod {
+	return &corev1.Pod{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Resources: corev1.ResourceRequirements{
+						Requests: *resources,
+						Limits:   *resources,
+					},
+				},
+			},
+		},
+	}
+}
+
+func makePodByResourceLists(resources ...corev1.ResourceList) *corev1.Pod {
+	pod := &corev1.Pod{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{},
+		},
+	}
+
+	for _, r := range resources {
+		pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+			Resources: corev1.ResourceRequirements{
+				Requests: r,
+				Limits:   r,
+			},
+		})
+	}
+
+	return pod
+}
+
+func makePodWithReqByResourceList(resources *corev1.ResourceList) *corev1.Pod {
```

Comment:
We can use functional options here instead https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis 
But probably in separate PR

Reply:
I tend to agree, but let's indeed defer to a separate PR.